### ISSUE TITLE
MeanSquaredError truncated to 32-bit precision when using 64-bit

### ISSUE THREAD

**Issue by jonwittmer** [tensorflow/tensorflow#39878]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Mint 19.2
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.3.0-dev20200526
- Python version: 3.8.3
- GPU model and memory: RTX 2080 Ti - 11GB

**Current Behavior**
 When computing the loss using `keras.losses.MeanSquaredError` with data being of `dtype=tf.float64,` the results are consistent with casting to float32 then back to float64. The result of calling the loss function is of `dtype=tf.float64,` but there is an error of about 1e-8. I have observed this behavior in versions 2.1, 2.2, and the nightly build indicated above. I have not tested this with other loss functions, although this may be an issue for them as well. 

**Expected Behavior** 
Calling the loss function should return the same results as computing the loss manually in 64-bits when the data is 64-bit. 

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. 
```
import tensorflow as tf
from tensorflow import keras

# some sample data 
x = tf.convert_to_tensor([[2.1]], dtype=tf.float64)
y_true = tf.square(x)

# sample output of NN - something not the same as y_true
y_pred = tf.convert_to_tensor([[3.68]], dtype=tf.float64)

# TF loss
loss = keras.losses.MeanSquaredError()
tf_loss = loss(y_pred, y_true)

# manually computed loss in 64-bit
man_loss64 = tf.square(y_pred - y_true)

# manually computed loss, cast to 32 bit, then back to 64
man_loss32 = tf.cast(tf.cast(tf.square(y_pred - y_true), dtype=tf.float32), dtype=tf.float64)

# difference between loss computations
diff64 = abs(man_loss64 - tf_loss)
diff32 = abs(man_loss32 - tf_loss)
print(f'Difference in 64 bit: {diff64}')
print(f'Difference in 32 bit: {diff32}')
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.1, 2.2 and nightly version(2.3.0-dev20200526) and was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/9cbaf9474dceb8c5184247e633a3bff5/untitled45.ipynb#scrollTo=T9IWe6lpeyO9).Thanks!

**Comment by Saduf2019** [comment]
I have tried in colab with TF 2.5 and was able to reproduce the issue.Please, find the [gist here](https://colab.research.google.com/gist/Saduf2019/e13a1e93aa24f9af370d05e2d067e60e/untitled589.ipynb).Thanks!

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
